index.ts

```diff
@@ -192,12 +205,37 @@ export class LinkedList<T> {
     const previousNode = targetNode.previous()
     previousNode.next = () => nextNode
     nextNode.previous = () => previousNode
+    targetNode.unlink()
     return (this.size -= 1)
   }
 
+  iterateOverList(
+    callback: NodeFunction<T>,
+    currentNode: Node<T> = this.head,
```

Comment:
after looking at this again there will always be a next node since the list cycles back around (the tail's next is the head and the head's previous is the tail) 

so what I really want to do is wrap the callback execution with a `if (currentIndex <= finalIndex)` so thanks for helping me find that!